tests/test_agenda_route.py

```diff
@@ -0,0 +1,139 @@
+from datetime import date, datetime, timedelta
+from app.database.models import User, Event
+
+
+HTTP_OK = 200
+
+
+class TestAgenda:
+    AGENDA = "/agenda"
+    AGENDA_7_DAYS = "/agenda?days=7"
+    AGENDA_30_DAYS = "/agenda?days=30"
+    NO_EVENTS = b"No events found..."
+    INVALID_DATES = b"Start date is greater than end date"
+
+    @staticmethod
+    def base_today_date():
+        return datetime.today().replace(hour=0, minute=0,second=0)
+
+    @staticmethod
+    def create_user_1(session):
+        user = User(username='zohar', email='aa.aa@aa.com', password='1234')
+        session.add(user)
+        session.commit()
+        return user
+
+    @staticmethod
+    def create_user_2(session):
+        user2 = User(username='dani', email='bb.aa@aa.com', password='12345')
+        session.add(user2)
+        session.commit()
+        return user2
+
+    @staticmethod
+    def add_event(session, title, content, start, end, owner_id):
+        event = Event(title=title, content=content, start=start, end=end, owner_id=owner_id)
+        session.add(event)
+        session.commit()
+
+    @staticmethod
+    def create_data_user_1(session):
+        user = TestAgenda.create_user_1(session)
+        base_date = TestAgenda.base_today_date()
+        # Today event
+        TestAgenda.add_event(session, "event 1", "...", base_date + timedelta(hours=7), base_date + timedelta(hours=9), user.id)
```

Comment:
Maybe you should use parameterize? :)

Reply:
I'm not sure parameterize suitable here, each test is for specific usage of the url, and each test needs some events to check which of them show up and which not.
therefore, I built this helper function for adding a group of 6 events for use in all the tests.